File: stack/Pilha.cs
using ListaEncadeada;
namespace Stack;
class Pilha : Lista {
  public Pilha() {}
  public void Push(string e) {
    No n = new(e);
     this.inserirNoInicio(n);
  }
  public No Pop() {
    No n = this.Buscar(1);
    this.deletarNaPos(1);
    return n;
  }
  public void Print()  => this.printar(); 
}

File: stack/pilha.go
package stack

import "aed/linkedList"

type Pilha struct {
	lista *linkedList.LinkedList
}

func NewPilha() *Pilha {
	return &Pilha{
		lista: linkedList.NewLinkedList(),
	}
}

func (p *Pilha) Push(element string) {
	p.lista.InsertAtPosition(1, element)
}

func (p *Pilha) Pop() string {
	No := p.lista.Search(1)
	p.lista.DeleteAtPosition(1)
	return No.Element()
}

func (p *Pilha) Printar() {
	p.lista.Print()
}
